reporthandling/results/v1/reportsummary/summarydetails.go

```diff
@@ -136,12 +136,26 @@ func (summaryDetails *SummaryDetails) ListResourcesIDs() *helpersv1.AllLists {
 	return summaryDetails.Controls.ListResourcesIDs()
 }
 
-// updateSummaryWithResource get the result of a single resource. If resource not found will return nil
+// AppendResourceResult appends the given resource result to the summary
+//
+// Updates any necessary info accordingly
 func (summaryDetails *SummaryDetails) AppendResourceResult(resourceResult *resourcesresults.Result) {
 
 	// update full-summary counter
 	updateControlsSummaryCounters(resourceResult, summaryDetails.Controls, nil)
 
+	// update the summary’s severity counters
+	if resourceResult.GetStatus(nil).IsFailed() {
+		for _, resourceControl := range resourceResult.ListControls() {
+			if resourceControl.GetStatus(nil).IsFailed() {
+				control := summaryDetails.Controls.GetControl(EControlCriteriaID, resourceControl.GetID())
+				severityScore := control.GetScoreFactor()
+				severity := apis.ControlSeverityToString(severityScore)
+				summaryDetails.SeverityCounters.Increase(severity, 1)
+			}
+		}
+	}
```

Comment:
I also chose to collect the severity counters when appending the resources, since we update control counters in this function anyway, yet it is generic enough in what it does.